### PR TITLE
Add new resources for 2022-06-01-preview

### PR DESCRIPTION
**Pull request by Juliehzl**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+          "Delete edge environment by name": {
+            "$ref": "./examples/connectedEnvironments_Delete.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "connectedEnvironments"
+        ],
+        "summary": "Update Edge Environment's properties.",
```
Comment (howang-ms) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json:258]: Edge => Azure Arc enabled

File: specification/app/resource-manager/readme.md [hunk 2]
```diff
@@ -40,12 +40,14 @@ input-file:
...
   - Microsoft.App/preview/2022-06-01-preview/ContainerApps.json
   - Microsoft.App/preview/2022-06-01-preview/ContainerAppsRevisions.json
-  - Microsoft.App/preview/2022-06-01-preview/DaprComponents.json
+  - Microsoft.App/preview/2022-06-01-preview/ManagedEnvironmentsDaprComponents.json
   - Microsoft.App/preview/2022-06-01-preview/Diagnostics.json
   - Microsoft.App/preview/2022-06-01-preview/Global.json
   - Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
   - Microsoft.App/preview/2022-06-01-preview/ManagedEnvironmentsStorages.json
   - Microsoft.App/preview/2022-06-01-preview/SourceControls.json
+  - Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json
```
Comment (howang-ms) [inline specification/app/resource-manager/readme.md:49]: - Microsoft.App/preview/2022-06-01-preview/AuthConfigs.json
  - Microsoft.App/preview/2022-06-01-preview/CommonDefinitions.json
  - Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json
  - Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsDaprComponents.json  
  - Microsoft.App/preview/2022-06-01-preview/ContainerApps.json
  - Microsoft.App/preview/2022-06-01-preview/ContainerAppsRevisions.json
  - Microsoft.App/preview/2022-06-01-preview/Diagnostics.json
  - Microsoft.App/preview/2022-06-01-preview/Global.json
  - Microsoft.App/preview/2022-06-01-preview/ManagedEnvironments.json
  - Microsoft.App/preview/2022-06-01-preview/ManagedEnvironmentsDaprComponents.json
  - Microsoft.App/preview/2022-06-01-preview/ManagedEnvironmentsStorages.json
  - Microsoft.App/preview/2022-06-01-preview/SourceControls.json

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Patch Managed Environment": {
+            "$ref": "./examples/ConnectedEnvironments_Patch.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/connectedEnvironments/{environmentName}/checkNameAvailability": {
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json:300]: >environmentName [](http://example.com/codeflow?start=119&length=15)

use {connectedEnvironmentName}.

The param names must be consistent

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsCertificates.json [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/connectedEnvironments/{environmentName}/certificates": {
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsCertificates.json:18]: >environmentName [](http://example.com/codeflow?start=119&length=15)

{connectedEnvironmentName}

Comment (ruslany) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsCertificates.json:18]: everywhere in this file

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsDaprComponents.json [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/connectedEnvironments/{environmentName}/daprComponents": {
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsDaprComponents.json:18]: >environmentName [](http://example.com/codeflow?start=119&length=15)

{connectedEnvironmentName}

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsStorages.json [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/connectedEnvironments/{environmentName}/storages": {
```
Comment (ruslany) [inline specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironmentsStorages.json:18]: >environmentName [](http://example.com/codeflow?start=119&length=15)

{connectedEnvironmentName}

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json (hunk 3 above)
Comment (ruslany) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json:300]: We took a last minute PR for 2022-03-01 to make all param names consistent. [https://github.com/Azure/azure-rest-api-specs/pull/19005](https://github.com/Azure/azure-rest-api-specs/pull/19005)

We should keep it that way as inconsistent names block terraform release.

File: specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json [hunk 7]
```diff
@@ -0,0 +1,449 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List environments by resource group": {
+            "$ref": "./examples/ConnectedEnvironments_ListByResourceGroup.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.App/connectedEnvironments/{name}": {
```
[the remark this answers is not in the document]
Comment (Juliehzl) [reply specification/app/resource-manager/Microsoft.App/preview/2022-06-01-preview/ConnectedEnvironments.json:100]: updated

**Review by ruslany — APPROVED** [review]
:shipit:

**Comment by Juliehzl** [comment]
@xiaoxuqi-ms could you help merge the PR?

**Comment by Juliehzl** [comment]
This is not breaking because the change is to align with server side behavior

**Review by NetRock — APPROVED** [review]

**Comment by Juliehzl** [comment]
@JeffreyRichter Could you help review the breaking change? But as I mentioned before, the change is to align with server side behavior, so it should not be breaking

**Comment by jijohn14** [comment]
@Juliehzl [Juliehzl](https://github.com/Juliehzl), Please update all examples of container apps for `environmentId`

**Pull request merged**